cmd/skopeo/proxy.go

```diff
@@ -266,6 +268,20 @@ func (h *proxyHandler) openImageImpl(args []any, allowNotFound bool) (replyBuf,
 		return ret, err
 	}
 
+	unparsedTopLevel := image.UnparsedInstance(imgsrc, nil)
+	policy, err := signature.DefaultPolicy(h.sysctx)
+	if err != nil {
+		return ret, err
+	}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return ret, err
+	}
+	allowed, err := policyContext.IsRunningImageAllowed(context.Background(), unparsedTopLevel)
+	if !allowed || err != nil {
+		return ret, err
```

Comment:
(A drive-by comment without fully reviewing the history.)

If `!allowed && err == nil`, this should probably fail (loudly about an unexpected state?), not return a success indication.

Reply:
We're not doing that in the other callees though AFAICS, but agree with the principle for security-relevant code.

But hmm, why does the API here not just return `error`?

Reply:
> (A drive-by comment without fully reviewing the history.)
> 
> If `!allowed && err == nil`, this should probably fail (loudly about an unexpected state?), not return a success indication.

OK updated

Reply:
> But hmm, why does the API here not just return `error`?

Just excessive paranoia. Looking at call sites like this one, I’m no longer sure this was a good tradeoff at all. But, well, we’re stuck with it.